librz/core/analysis_tp.c

```diff
@@ -564,6 +589,12 @@ void propagate_types_among_used_variables(RzCore *core, HtUP *op_cache, Sdb *tra
 	bool userfnc = false;
 	bool prop = false;
 	ut32 type = aop->type & RZ_ANALYSIS_OP_TYPE_MASK;
+
+	RzILTraceInstruction *cur_instr_trace = NULL;
```

Comment:
I would just pass `RzEsilTrace` to this function instead of RzPVector and then use `esil_trace_get_instruction(etrace, idx);` to get the right instruction. By doing so all the proper checks are in that function and not here.

Reply:
It also allow us to easily change the implementation of how traced instructions are stored without having to touch again all this code.

Reply:
Yeah, there is a similar get function, but I don't use it sometimes.  I will update them to prevent getting element directly from the vector.